src/parsing.rs

```diff
@@ -84,9 +91,12 @@ pub fn try_parse_request(buffer: Vec<u8>) -> Result<ParseResult, httparse::Error
                     path: slice_indices(&*buffer, r.path.unwrap().as_bytes()),
                 };
 
-                (r, proto, n)
+                let method = slice_indices(&*buffer, r.method.unwrap().as_bytes());
```

Comment:
I am not sure about this unwrap, but the `path`-parsing is using it, too. So I just did the same here.